yearn/README.adoc

```diff
@@ -0,0 +1,181 @@
+optional extra reward tokens are used to buy wBTC via a
+decentralized exchange. At the end, the strategy takes acquired wBTC
+and deposits them to the Curve tBTC v2 pool. This way it obtains new
+LP tokens the vault is interested for, and makes the profit in
+result. At this stage, the strategy may repay some debt back to the
+vault, if needed. The entire cycle repeats for the strategy lifetime
+so all gains are constantly reinvested. Worth to flag that current
+implementation uses wBTC as the intermediary token because
+of its liquidity and ubiquity in BTC-based Curve pools.
+
+Long story short, this strategy is a more sophisticated version of the
+`CurveVoterProxyStrategy` because it uses Convex as intermediary instead
+of interacting with Curve directly. In result, it obtains CVX tokens which
+increase their profit. However, this strategy incurs additional fees and
+distributes extra rewards using a different model. This should be taken
+into account when comparing profitability of different strategies.
+
+=== `SaddleStrategy` contract
+
+Currently implemented. Will be described soon.
+
+== Build, test and deploy
+
+Yearn components use https://hardhat.org/[*Hardhat*] development environment.
+To build and deploy these contracts, please follow the instructions presented
+below.
+
+=== Prerequisites
+
+Please make sure you have the following prerequisites installed on your machine:
+
+- https://nodejs.org[Node.js] >12.22.1
+- https://yarnpkg.com[Yarn] >1.22.10
+
+=== Build contracts
+
+To build the smart contracts, install node packages first:
+```
+yarn install
+```
+Once packages are installed, you can build the smart contracts using:
+```
+yarn build
+```
+Compiled contracts will land in the `build` directory.
+
+=== Test contracts
+
+There are multiple system test scenarios living in the `test/system` directory.
+They all leverage the Hardhat mainnet forking feature which allow to test
+contract using mainnet state.
+You can run them by doing:
+```
+FORKING_URL=<url> yarn test:system
+```
+where `FORKING_URL` is an URL of an Ethereum mainnet node with access to
+archival data (Alchemy recommended)
+
+=== Deploy contracts
+
+To deploy all contracts on the given network, please run:
+```
+yarn deploy --network <network>
+```
+with required environment variables listed in the <<deployment-parametrization>>
+section.
+
+To deploy a single contract, use the same command with the `--tags` option
+followed by the contract name.
+
+If contracts haven't been built yet or changes occurred, this task will build
+the contracts before running the deployment script. This command produces
+an `export.json` file containing contract deployment info.
+
+Please note that all strategies contracts make a call to their vault contract
+upon initialization. That means the deployment will fail if the vault address
+passed in the strategy constructor does not point to a contract compatible
+with the Yearn vault API.
+
+[[deployment-parametrization]]
```

Comment:
Was there a specific reason on why using addresses of contracts as env params? Do you think we can follow the path of deployment as we have in [coverage-pool](https://github.com/keep-network/coverage-pools/blob/main/README.adoc#predefined-artifacts) and having "./external/testnet" & "./external/main" pattern for the addresses? I think the current solution is fine, but I think we want to minimize a human factor which might result in a typo, copy&paste error or chosing a wrong contract during deployment.
If you'd agree, then the deployment scripts will need to be adjusted as well.
If there are no contracts available for "./external/testnet" then I guess we can use only "./external/main" and replace the addresses once tBTCv2 & new pools are deployed to mainnet.

Reply:
The main cause was the fact that most of the external dependencies are not shipped as npm packages and we don't know their addresses until tBTC v2 is deployed. However, your proposition sounds good and we can use `Predefined artifacts` with zero addresses for now and leave TODOs to fill proper addresses once we know them.

Reply:
See https://github.com/keep-network/tbtc-v2/pull/14/commits/0e577c4a3dcf2c3cb8cc20520e537b39d3c8d18e